File: LeetCode75_LevelOne/Best_Time_To_Buy_and_Sell_Stock/Best_Time_to_Biu_and_Sell_Stock.cc
// 121. Best Time to Buy and Sell Stock C++
#include <algorithm>
#include <vector>

class Solution {
public:
    int maxProfit(std::vector<int>& prices) {
        int left = 0;
        int right = 1;
        int profit = 0;
        int length = prices.size();
        while (right < length) {
            int curr = prices.at(right) - prices.at(left);
            if (curr > 0) {
                profit = std::max(curr, profit);
            } else {
                left = right;
            }
            right++;
        }
        return profit;
    }
};

File: LeetCode75_LevelOne/Best_Time_To_Buy_and_Sell_Stock/Best_Time_to_Biu_and_Sell_Stock.py
# 121. Bset Time to Buy and Sell Stock Python

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        left = 0
        right = 1
        profit = 0
        while right < len(prices):
            currentProfit = prices[right] - prices[left] #our current Profit
            if prices[left] < prices[right]:
                profit =max(currentProfit,profit)
            else:
                left = right
            right += 1
        return profit
